cmd/checks_windows.go

```diff
@@ -294,13 +294,11 @@ func userInGroup(userName, groupName string) (bool, error) {
 }
 
 func firewallUp() (bool, error) {
-	fwProfiles := []string{"Domain", "Public", "Private"}
-	for _, profile := range fwProfiles {
-		// This is kind of jank and kind of slow
-		cmdText := "(Get-NetFirewallProfile -Name '" + profile + "').Enabled"
-		result, err := commandOutput(cmdText)
-		if result != "True" || err != nil {
-			return false, err
+	fwProfilesInt := []int{wapi.NET_FW_PROFILE2_DOMAIN, wapi.NET_FW_PROFILE2_PRIVATE, wapi.NET_FW_PROFILE2_PUBLIC}
+	for profile := range fwProfilesInt {
+		profileResult, err := wapi.FirewallIsEnabled(int32(profile))
+		if err != nil || profileResult == false {
```

Comment:
We probably want to handle the `err != nil` and `profileResult == false` cases separately. AFAIK if there's an error returned from that call, it's likely because there was an issue retrieving the property value [here](https://github.com/iamacarpet/go-win64api/blob/fe38760bed28/firewall.go#L470), in which case we should return an error as well so that the check never succeeds.

Reply:
What's the point though? In the case that `err != nil`, you still return `false`, so the check won't succeed regardless

Reply:
I think if the check was `FirewallUpNot` (even though I'm not sure what the use case for that would be...), then the check would succeed if for some reason the library was unable to initialize or retrieve values through the WinAPI, whereas if we return an error then it will never award points.

Reply:
Ah, I see